src/main/java/frc/robot/subsystems/PoseEstimation.java

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import java.util.List;
+import java.util.Optional;
+import org.photonvision.EstimatedRobotPose;
+import org.photonvision.targeting.PhotonTrackedTarget;
+import edu.wpi.first.math.estimator.SwerveDrivePoseEstimator;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj.smartdashboard.Field2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.SwerveConfig;
+import frc.robot.constants.CompConstants;
+import frc.robot.constants.VisionConstants;
+
+
+
+public class PoseEstimation extends SubsystemBase{
+  private  Limelight _limelight;
+  private  SwerveSubsystem _swerve;
+  private  SwerveDrivePoseEstimator _poseEstimator;
+  private  Field2d _field = new Field2d();
+
+  private boolean _hasUpdated = false; 
+
+  public PoseEstimation(Limelight lime, SwerveSubsystem swerveSubsystem) {
+    _limelight = lime;
+    _swerve = swerveSubsystem;
+    
```

Comment:
In general in a code base, unless there's a good reason, it helps to use common names for common things. Here you have fields called linelight and swerve, but parameters called lime and swerveSubsystem.

Reply:
I tried in the last few commits to follow all the sonarLint naming conventions. Now when I am defining an instance variable to an object provided in the constructor, I am naming the object provided in the constructor "tmp<instance variable name>" for the sake of consistency.